### ISSUE TITLE
[Feature] Show days on Xaxis that are not in the dataset

### ISSUE THREAD

**Issue by slavonicsniper** [apache/echarts#19488]
### What problem does this feature solve?

I would like to be able to show all days on the xAxis, even these that do not have an entry in the data. Is that already possible via an option or do I need to add them manually into the data array with 0 value?

### What does the proposed API look like?

An option for showing all days on the xAxis

**Comment by MatthiasMert** [comment]
An example would be helpful to clear up any uncertainty about your use case. 

As it is I would suggest taking a look at [interval](https://echarts.apache.org/en/option.html#xAxis.interval) or [minInterval ](https://echarts.apache.org/en/option.html#xAxis.minInterval)/ [maxInterval](https://echarts.apache.org/en/option.html#xAxis.maxInterval)

**Comment by slavonicsniper** [comment]
['01-01-2024', '03-01-2024', '05-01-2024'] -> dataList for xAxis, I will see only these specific dates, is there a way to show the other dates like '02-01-2024' on the xAxis without adding them manually to the dataList?

**Comment by MatthiasMert** [comment]
Still not a very elaborate example. But from that I am now guessing that you are using `axis type 'category'`. [Here ](https://echarts.apache.org/examples/en/editor.html?c=line-simple&code=MYewdgzgLgBAJgQygmBeGBtDByATABlwBYBafARjPOwBoZyBWfAXTpwOKrIGZaZduLNnkKkKZBn3IAOIe1FdyvOuQLNmAbgBQWkAAcoAS3BoYAby0wYADwCC1wxABc5y1ZhQAnnoCmL7EYAtj60bgC-NG6e9o4uFu4e3n4w2ABuCAA2AK4h4ZFWED4AToY-zphuVvEJib7-GYZgIfk1iMgubQgtVmFuzFphGkA)is an example using `axis type 'time'`.

**Comment by slavonicsniper** [comment]
![image](https://github.com/apache/echarts/assets/22478414/7e67aa18-c2aa-4035-ac55-60c998625781)

basically my idea was to have a point for days which are not in the data and to assign them value 0 but I guess that's not possible unless I manually create entries with 0 value in the data array

**Comment by MatthiasMert** [comment]
I dont understand why the solution I provided is not applicable. You can also use the [label formatter function](https://echarts.apache.org/en/option.html#xAxis.axisLabel.formatter) to bring the dates into a format of your choice: [Example](https://echarts.apache.org/examples/en/editor.html?c=line-simple&code=MYewdgzgLgBAJgQygmBeGBtDByATABlwBYBafARjPOwBoZyBWfAXTpwOKrIGZaZduLNnkKkKZBn3IAOIe1FdyvOuQLNmAbgBQWkAAcoAS3BoYAby0wYADwCC1wxABc5y1ZhQAnnoCmL7EYAtj60blYIDhAAMggARj4ANi4W7u4QABYgAO4AshEx8UkeAE4Arj40Ye4AZiDFgUhQPsUuUCAAIkg-AMpQxYZgAOZVAL5uI5VWnvaOyVVevv4AbggJ5djjkzAQzYY-zphVKake3n4w2AkDIVupiMgu9wi3MGNWzFoj2jrVpWDARhMbU6TV6_SGAAo9AhighAhAAJSuKygSCwe4-UxgHxZGAgnxQmFwxHaKzFHxQUrFMDwLoAOkGFPxEKRAGoLiRsDB2RiGRScuAoOkWdz6KLsJzRbzGVAAGKlBIJACaPhhLO0IyAA)

**Comment by slavonicsniper** [comment]
In your solution, only dates that are in const data are displayed, ['01-01-2024', '03-01-2024', '05-01-2024']

I would like to show all the missing dates from 01-01-2024 to 05-01-2024 like 02-01 or 04-01 with value '0'. So I was wondering whether there's an option to do that but it doesn't look like there is.

**Comment by helgasoft** [comment]
> an option to do that but it doesn't look like there is.

Correct, ECharts, like any other charting software, does not create data points, just displays them.

NB: 📌 please close issue if problem solved.

**Comment by Alia2006** [comment]
I am in bar type

**Comment by pasch82-code** [comment]
Echarts is not able to understand a missing time point with "line" type, it will try to connect the line.

The ‘bar’ type is not affected by this limitation.
![image](https://github.com/user-attachments/assets/5378cdea-79e8-463c-93b1-5d4776e51cc0)
